cookbook/examples/customer_support/workflow.py

```diff
@@ -0,0 +1,137 @@
+    )
+
+    def __init__(self):
+        super().__init__()
+        setup_database()
+
+    @Workflow.register(description="Order a product for a user")
+    def order_product(self, user_email: str, product_id: int) -> str:
+        response = self.database_agent.run(f"Order a product for user {user_email} with product id {product_id}")
+        return str(response.content)
+
+    @Workflow.register(description="Get the product id of an order")
+    def get_product_id(self, product_name: str) -> str:
+        response = self.database_agent.run(f"Get the product id of product {product_name}")
+        return str(response.content)
+
+    @Workflow.register(description="Get order status")
+    def get_order_status(self, order_id: int) -> str:
+        response = self.database_agent.run(f"Get the status of order {order_id}")
+        return str(response.content)
+
+    @Workflow.register(description="Get all orders for a user")
+    def get_all_orders(self, user_email: str) -> str:
+        response = self.database_agent.run(f"Get all orders for user {user_email}")
+        return str(response.content)
+
+    @Workflow.register(description="Get the latest order for a user")
+    def get_latest_order(self, user_email: str) -> str:
+        response = self.database_agent.run(f"Get the latest order for user {user_email}")
+        return str(response.content)
+
+    @Workflow.register(description="Return an order")
+    def return_order(self, order_id: int, user_email: str) -> str:
+        response = self.database_agent.run(
+            f"Return order {order_id} for user {user_email} if the order status is COMPLETED else cancel the order"
+        )
+        return str(response.content)
+
+    @Workflow.register(description="Cancel an order")
+    def cancel_order(self, order_id: int, user_email: str) -> str:
+        response = self.database_agent.run(
+            f"Cancel order {order_id} for user {user_email} if the order status is ORDERED or DISPATCHED"
+        )
+        return str(response.content)
+
+    @Workflow.register(description="Handle a general query")
+    def handle_general_query(self, query: str) -> str:
+        response = self.general_agent.run(f"Handle the general query: {query}")
+        return str(response.content)
+
+    @Workflow.register(description="Get the details of a product")
+    def get_product_details(self, product_id: int) -> str:
+        response = self.database_agent.run(f"Get the details of product {product_id}")
+        return str(response.content)
+
+    @Workflow.register(description="Get the details of an order")
+    def get_order_details(self, order_id: int) -> str:
+        response = self.database_agent.run(f"Get the details of order {order_id}")
+        return str(response.content)
+
+
+agent = Agent(
+    name="Customer Support Agent",
+    description="You are a customer support agent that can help with general queries and order products for users",
+    instructions=[
```

Comment:
Couldn't this example be done with only the database agent and structured output?